### PR TITLE
k8s-bake-update

### PR DESCRIPTION
**Pull request by rsamigullin**
PR is ready for review

### CODE REVIEW

File: src/helm-util.ts
Comment (OliverMKing) [inline src/helm-util.ts:111]: use const here. if you can't use const when declaring a variable you should pretty much always use let.

File: src/helm-util.ts
Comment (OliverMKing) [inline src/helm-util.ts:110]: use let

File: src/kompose-util.ts
Comment (OliverMKing) [inline src/kompose-util.ts:19]: use const

File: src/kompose-util.ts
Comment (OliverMKing) [inline src/kompose-util.ts:21]: version should be refactored out to a constant variable somewhere. It's used a few times it looks like.

File: src/kubectl-util.ts
Comment (OliverMKing) [inline src/kubectl-util.ts:68]: use const

**Review by OliverMKing — COMMENTED** [review]
Nice changes! Should use const whenever possible then let. Var shouldn't really ever be used.

**Comment by OliverMKing** [comment]
You should also add yourself as the codeowner in the .github folder

File: .github/CODEOWNERS
Comment (OliverMKing) [inline .github/CODEOWNERS:1]: Just remove them because they no longer own this action. You can add me instead. @OliverMKing

File: src/helm-util.ts
Comment (OliverMKing) [inline src/helm-util.ts:56]: Use a default parameter for this

File: src/helm-util.ts
Comment (OliverMKing) [inline src/helm-util.ts:113]: "latest" should be a constant that is used everywhere

File: src/run.ts
Comment (OliverMKing) [inline src/run.ts:210]: Just do:

const isSilent = core.getInput('silent') === 'false';

The three = checks for type equality which allows for it to be condensed this way.

Core.getInput defaults to not required so this just cuts down some unnecessary things.

File: src/utilities.ts
Comment (OliverMKing) [inline src/utilities.ts:33]: why not just set this when you declare execResult?

**Review by OliverMKing — CHANGES_REQUESTED** [review]
I think this is a good start. After this PR is merged in I think even more can be refactored. The code for baking with different tools is largely the same and there is a lot of repeated logic. We can use duck typing like polymorphism to make things a lot cleaner overall

File: src/utilities.ts
Comment (OliverMKing) [inline src/utilities.ts:60]: small nitpick but constants like this are usually capitalized like LATEST. When it's something that is used periodically and is a constant value.

**Review by OliverMKing — APPROVED** [review]

File: lib/helm-util.js
Comment (davidgamero) [inline lib/helm-util.js:70]: is it possible to just use `export function` instead of having a separate line for setting the module exports?

File: src/run.ts
Comment (davidgamero) [inline src/run.ts:185]: i think this logic will throw error on higher major versions with minor<14 ex: 2.13

Comment (davidgamero) [reply src/run.ts:185]: maybe add unit tests for checking this to make sure?

File: lib/helm-util.js
Comment (rsamigullin) [reply lib/helm-util.js:70]: @davidgamero looks like it's not.
I checked docs for tsconfig compiler and there is no option for that :
https://www.typescriptlang.org/tsconfig

File: src/helm-util.ts
Comment (davidgamero) [inline src/helm-util.ts:54]: can this be switched to `export function walkSync(dir, filelist = [], fileToFind) {` to match the other functions in the file?

File: lib/helm-util.js
Comment (davidgamero) [reply lib/helm-util.js:70]: okay gotcha

File: src/run.ts
Comment (davidgamero) [reply src/run.ts:185]: `!clientVersion || (clientVersion.major <=1 && clientVersion.minor < 14)`

**Review by davidgamero — COMMENTED** [review]
Looks good!

**Review by davidgamero — APPROVED** [review]

**Pull request merged**
